src/main/js/ephox/imagetools/transformations/ImageTools.js

```diff
@@ -80,13 +82,13 @@ define("ephox/imagetools/transformations/ImageTools", [
 
   function resize(blob, w, h) {
     return Conversions.blobToImage(blob).then(function(image) {
-      var canvas = Canvas.create(w, h),
-        context = Canvas.get2dContext(canvas);
-
-      context.drawImage(image, 0, 0, w, h);
-      revokeImageUrl(image);
-
-      return Conversions.canvasToBlob(canvas, blob.type);
+      return ImageResizerWebgl.scale(image, w, h)
```

Comment:
Hm... algorithm is not something that might fail, so fallback sequence probably won't work here. I found good visual comparison of some image resampling algorithms [here](https://en.wikipedia.org/wiki/Image_scaling), with explanations.

Choosing algorithm base on image format and contents is interesting idea - we could use bilinear or lanczos for pictures and for example hqx for screenshots of texts or graphics. Although not sure how we could automate this. Probably it is better if the end user simply passes in particular implementation.